### PR TITLE
Hf/audio slider connected to volume

### PR DESCRIPTION
**Pull request by OliverGlandberger**

### CODE REVIEW

File: Sail/src/API/Audio/AudioEngine.cpp [hunk 1]
```diff
@@ -632,7 +634,7 @@ HRESULT AudioEngine::initXAudio2() {
 	UINT32 flags = XAUDIO2_1024_QUANTUM;
 
 #ifdef DEVELOPMENT
-	flags |= XAUDIO2_DEBUG_ENGINE;
+	//flags |= XAUDIO2_DEBUG_ENGINE;
 #endif
```
Comment (soridanm) [inline Sail/src/API/Audio/AudioEngine.cpp]: Is it necessary to #ifdef when not doing anything in the loop?

File: Sail/src/API/Audio/AudioEngine.cpp [hunk 2]
```diff
@@ -653,7 +655,7 @@ HRESULT AudioEngine::initXAudio2() {
 
 #if DEVELOPMENT
 	// Activate debug layer if we're in development
-	activateDebugLayer();
+	//activateDebugLayer();
 #endif
```
Comment (soridanm) [inline Sail/src/API/Audio/AudioEngine.cpp]: Same thing here

File: Sail/src/Sail/entities/systems/Audio/AudioSystem.cpp [hunk 3]
```diff
@@ -183,7 +183,7 @@ void AudioSystem::update(Camera& cam, float dt, float alpha) {
 							randomSoundIndex = randomASoundIndex(soundPoolSize, soundGeneral);
 
 							soundUnique = &audioData.m_soundsUnique[soundTypeIndex].at(randomSoundIndex);
-							soundGeneral->volume = soundUnique->volume;
+							soundGeneral->volume = (soundUnique->volume * Application::getInstance()->getSettings().applicationSettingsDynamic["sound"]["global"].value);
```
Comment (soridanm) [inline Sail/src/Sail/entities/systems/Audio/AudioSystem.cpp]: Replace Application::getinstance()->getSettings() with m_appSettings?

File: Sail/src/API/Audio/AudioEngine.cpp (hunk 1 above)
Comment (OliverGlandberger) [reply Sail/src/API/Audio/AudioEngine.cpp]: Yes box

File: Sail/src/API/Audio/AudioEngine.cpp (hunk 2 above)
Comment (OliverGlandberger) [reply Sail/src/API/Audio/AudioEngine.cpp]: Yes box

**Pull request merged**